app/utilities/metadata_parser.py

```diff
@@ -146,6 +146,32 @@ def convert_schema_name(self, data, **kwargs):
             )
         return data
 
+    @post_load
+    def update_response_id(
+        self, data, **kwargs
+    ):  # pylint: disable=no-self-use, unused-argument
+        """
+        If response_id is present : return as it is
+        If response_id is not present : Build response_id from ru_ref,collection_exercise_sid,eq_id and form_type
+                                        and updates metadata with response_id
+
+
+        """
+        if data.get("response_id"):
+            return data
```

Comment:
We should add a log line here to say something like "respose_id exists in claim, skipping generation". That way, it is easy to view in logs which path it took.

Reply:
But don't write out the `response_id` anywhere in the log. Even though this is now going to be using known values in the interim, we should continue to treat it as a secret.